src/type_checker/mod.rs

```diff
@@ -49,15 +49,17 @@ impl FullyQualified {
         Self { module: None, name }
     }
 
-    pub fn new(module: &ModulePath, name: &String) -> Self {
+    // TODO: Pass in `String`, instead of `&str`, so we don't hide an
+    // allocation within.
+    pub fn new(module: &ModulePath, name: &str) -> Self {
         let module = match module {
             ModulePath::Local => None,
             ModulePath::Alias(_) => unreachable!(),
             ModulePath::Absolute(user_repo) => Some(user_repo.clone()),
         };
         Self {
             module,
-            name: name.clone(),
+            name: name.to_string(),
```

Comment:
imo `name` should be passed as a `String`, so that we don't do a clone internally here.